### ISSUE TITLE
Error in a relationship projection containing double quotes

### ISSUE THREAD

**Issue by BertrandFritsch** [neo4j/graphql#600]
**Describe the bug**
An error occurs when there are quotes in a relationship projection in a subquery.

The GraphQL query:
```graphql
query Document {
  stakeholders {
    documents {
      customerContactConnection {
        edges {
          fileId
          uploadedAt
        }
      }
    }
  }
}
```

Here is the generated Cypher query:
```cypher
MATCH (this:Stakeholder)
RETURN this { documents: [ (this)-[:REQUIRES]->(this_documents:Document)   | this_documents { customerContactConnection: apoc.cypher.runFirstColumn("CALL {
WITH this_documents
MATCH (this_documents)<-[this_documents_uploaded_relationship:UPLOADED]-(this_documents_customercontact:CustomerContact)
WITH collect({ fileId: this_documents_uploaded_relationship.fileId, uploadedAt: apoc.date.convertFormat(toString(this_documents_uploaded_relationship.uploadedAt), "iso_zoned_date_time", "iso_offset_date_time") }) AS edges
RETURN { edges: edges, totalCount: size(edges) } AS customerContactConnection
} RETURN customerContactConnection", { this_documents: this_documents }, false) } ] } as this
```
The result:
```json
{
  "errors": [
    {
      "message": "Invalid input 'iso_zoned_date_time': expected\n  \"!=\"\n  \"%\"\n  \")\"\n  \"*\"\n  \"+\"\n  \",\"\n  \"-\"\n  \".\"\n  \"/\"\n  \":\"\n  \"<\"\n  \"<=\"\n  \"<>\"\n  \"=\"\n  \"=~\"\n  \">\"\n  \">=\"\n  \"AND\"\n  \"CONTAINS\"\n  \"ENDS\"\n  \"IN\"\n  \"IS\"\n  \"OR\"\n  \"STARTS\"\n  \"XOR\"\n  \"[\"\n  \"^\" (line 5, column 165 (offset: 486))\n\"WITH collect({ fileId: this_documents_uploaded_relationship.fileId, uploadedAt: apoc.date.convertFormat(toString(this_documents_uploaded_relationship.uploadedAt), \"iso_zoned_date_time\", \"iso_offset_date_time\") }) AS edges\"\n                                                                                                                                                                     ^",
      "locations": [
        {
          "line": 2,
          "column": 3
        }
      ],
      "path": [
        "stakeholders"
      ],
      "extensions": {
        "code": "INTERNAL_SERVER_ERROR",
        "exception": {
          "code": "Neo.ClientError.Statement.SyntaxError",
          "name": "Neo4jError",
          "stacktrace": [
            "Neo4jError: Invalid input 'iso_zoned_date_time': expected",
            "  \"!=\"",
            "  \"%\"",
            "  \")\"",
            "  \"*\"",
            "  \"+\"",
            "  \",\"",
            "  \"-\"",
            "  \".\"",
            "  \"/\"",
            "  \":\"",
            "  \"<\"",
            "  \"<=\"",
            "  \"<>\"",
            "  \"=\"",
            "  \"=~\"",
            "  \">\"",
            "  \">=\"",
            "  \"AND\"",
            "  \"CONTAINS\"",
            "  \"ENDS\"",
            "  \"IN\"",
            "  \"IS\"",
            "  \"OR\"",
            "  \"STARTS\"",
            "  \"XOR\"",
            "  \"[\"",
            "  \"^\" (line 5, column 165 (offset: 486))",
            "\"WITH collect({ fileId: this_documents_uploaded_relationship.fileId, uploadedAt: apoc.date.convertFormat(toString(this_documents_uploaded_relationship.uploadedAt), \"iso_zoned_date_time\", \"iso_offset_date_time\") }) AS edges\"",
            "                                                                                                                                                                     ^",
            "",
            "    at captureStacktrace (C:\\Users\\bertr\\work\\projects\\portal-api\\node_modules\\.pnpm\\neo4j-driver-core@4.4.0\\node_modules\\neo4j-driver-core\\lib\\result.js:239:17)",
            "    at new Result (C:\\Users\\bertr\\work\\projects\\portal-api\\node_modules\\.pnpm\\neo4j-driver-core@4.4.0\\node_modules\\neo4j-driver-core\\lib\\result.js:59:23)",
            "    at newCompletedResult (C:\\Users\\bertr\\work\\projects\\portal-api\\node_modules\\.pnpm\\neo4j-driver-core@4.4.0\\node_modules\\neo4j-driver-core\\lib\\transaction.js:375:12)",
            "    at Object.run (C:\\Users\\bertr\\work\\projects\\portal-api\\node_modules\\.pnpm\\neo4j-driver-core@4.4.0\\node_modules\\neo4j-driver-core\\lib\\transaction.js:229:20)",
            "    at Transaction.run (C:\\Users\\bertr\\work\\projects\\portal-api\\node_modules\\.pnpm\\neo4j-driver-core@4.4.0\\node_modules\\neo4j-driver-core\\lib\\transaction.js:101:34)",
            "    at C:\\Users\\bertr\\work\\projects\\portal-api\\node_modules\\.pnpm\\@neo4j+graphql@2.4.0_4751c90ed1bf2b5f1fc2484aa9f1a4b9\\node_modules\\@neo4j\\graphql\\src\\utils\\execute.ts:98:35",
            "    at TransactionExecutor._safeExecuteTransactionWork (C:\\Users\\bertr\\work\\projects\\portal-api\\node_modules\\.pnpm\\neo4j-driver-core@4.4.0\\node_modules\\neo4j-driver-core\\lib\\internal\\transaction-executor.js:92:26)",
            "    at TransactionExecutor._executeTransactionInsidePromise (C:\\Users\\bertr\\work\\projects\\portal-api\\node_modules\\.pnpm\\neo4j-driver-core@4.4.0\\node_modules\\neo4j-driver-core\\lib\\internal\\transaction-executor.js:83:34)",
            "    at C:\\Users\\bertr\\work\\projects\\portal-api\\node_modules\\.pnpm\\neo4j-driver-core@4.4.0\\node_modules\\neo4j-driver-core\\lib\\internal\\transaction-executor.js:40:19",
            "    at new Promise (<anonymous>)"
          ]
        }
      }
    }
  ],
  "data": null
}
```
**Type definitions**
```graphql
interface UploadedDocument @relationshipProperties {
  fileId: ID!
  uploadedAt: DateTime!
}

type Document @exclude(operations: [CREATE, UPDATE, DELETE]) {
  id: ID! @id
  stakeholder: Stakeholder! @relationship(type: "REQUIRES", direction: OUT)

  customerContact: CustomerContact @relationship(type: "UPLOADED", properties: "UploadedDocument", direction: IN)
}

type CustomerContact @exclude(operations: [CREATE, UPDATE, DELETE]) {
  email: String!
  firstname: String!
  lastname: String!
  documents: [Document!]! @relationship(type: "UPLOADED", properties: "UploadedDocument", direction: OUT)
}

type Stakeholder @exclude(operations: [CREATE, UPDATE, DELETE]) {
  id: ID!
  fields: String!
  documents: [Document!]! @relationship(type: "REQUIRES", direction: OUT)
}
```
**Expected behavior**

The quotes should be escaped.

**System (please complete the following information):**

-   OS: Windows
-   Version: @neo4j/graphql@2.4.0
-   Node.js version: 17.0.1

**Comment by darrellwarde** [comment]
Hey @BertrandFritsch, I believe the bug you are describing here has been fixed in #587, and will be included in the next release! 🙂  Closing for now.

**Comment by danielfein** [comment]
Hi @darrellwarde do you know when the fix for this one might be added?
